easybuild/tools/module_generator.py

```diff
@@ -85,32 +85,43 @@ def module_load_regex(modfilepath):
     return re.compile(regex, re.M)
 
 
-def dependencies_for(mod_name, modtool, depth=None):
+def dependencies_for(mod_name, modtool, depth=None, alldeps=None, full_module_names=False):
     """
     Obtain a list of dependencies for the given module, determined recursively, up to a specified depth (optionally)
+    :param alldeps: recursion depth (default is None, which corresponds to infinite recursion depth)
     :param depth: recursion depth (default is None, which corresponds to infinite recursion depth)
+    :param full_module_names: use $MODULEPATH to build filepath instead of get_value_from_modulefile
     """
-    mod_filepath = modtool.modulefile_path(mod_name)
+
+    if alldeps is None:
+        alldeps = {}
+
+    if mod_name not in alldeps:
+        alldeps[mod_name] = []
```

Comment:
No need for this if you use `alldeps.setdefault(mod_name, []).extend(newdeps)` below